owners/src/repo/virtual_repo.js

```diff
@@ -58,6 +77,14 @@ module.exports = class VirtualRepository extends Repository {
    * @param {?function} cacheMissCallback called when there is a cache miss.
    */
   async warmCache(cacheMissCallback) {
+    // The GitHub Search API is non-deterministic, and will frequently return
+    // duplicates or leave out files. It syncs periodically so if a given search
+    // drops one or two files, it usually won't cause any problems. When
+    // starting up the virtual repo and warming the cache, however, we want to
+    // make sure we know all the files we need to know about, so we try a couple
+    // times.
+    await Promise.all([this.sync(), this.sync(), this.sync()]);
```

Comment:
I... I don't know how to feel about this 😐

Reply:
I don't like it at all. It's what's currently there, I don't really want to check it in but I put it here cause it's what currently did the job. Very open to ideas

Reply:
For now, removed this hack and restored the sync-on-read approach. It will do the job for now, until I have the chance to implement more resilient caching/initialization